.github/workflows/dry-run.yml

```diff
@@ -0,0 +1,33 @@
+name: dry-run
+
+on:
+  push:
+    branches: [develop]
+    paths:
+      - org/**
+  workflow_dispatch:
+    inputs:
+      force:
+        type: boolean
+        required: false
+
+jobs:
+  lint-yaml:
+    uses: ./.github/workflows/_lint.yml
+  dry-run:
+    needs: lint-yaml
+    uses: ./.github/workflows/_sync.yml
+    permissions: read-all
+    secrets: inherit
+    with:
+      confirm: false
+  sync-branch:
+    needs: dry-run
+    runs-on: ubuntu-latest
+    steps:
+      - name: 🔄 sync branch
+        uses: connor-baer/action-sync-branch@v1.0.3
+        with:
+          token: ${{ secrets.SERVICE_ACCOUNT_TOKEN }}
+          branch: main
+          force: false
```

Comment:
Might not need this if we can run validation workflows (which could include linting and a dry-run) on fork PRs _once they've been approved_